crates/interpreter/src/opcode.rs

```diff
@@ -276,43 +284,132 @@ impl OpCode {
 }
 
 /// Information about opcode, such as name, and stack inputs and outputs.
-#[derive(Clone, Copy, Default, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct OpCodeInfo {
-    pub name: &'static str,
-    pub inputs: u8,
-    pub outputs: u8,
-    // TODO make this a bitfield
-    pub is_eof: bool,
-    // If the opcode is return from execution. aka STOP,RETURN, ..
-    pub is_terminating_opcode: bool,
-    /// Size of opcode with its intermediate bytes.
-    ///
-    /// RJUMPV is special case where the bytes len is depending on bytecode value,
-    /// for RJUMV size will be set to one byte while minimum is two.
-    pub immediate_size: u8,
+    // TODO: make this a bitfield. This is OK for now since we have a few alignment bytes to spare.
+    // Invariant: `(name_ptr, name_len)` is a `&'static str`.
+    name_ptr: NonNull<u8>,
+    name_len: u8,
+    inputs: u8,
+    outputs: u8,
+    not_eof: bool,
+    terminating: bool,
+    imm_size: u8,
```

Comment:
Would revert to `immediate_size`, imm can be confusing. And revert comment's

Reply:
"imm" is a common abbreviation for "immediate" in bytecodes/VMs/compilers.
Comments were moved to doccomments in the related methods.

Reply:
immediate is more readable for devs that are checking code.